experimental/AlgebraicStatistics/src/DiscreteGraphicalModels.jl

```diff
@@ -0,0 +1,326 @@
+graphical model. This is the same as the `global_markov` property of the underlying graph.
+"""
+function ci_structure(M::DiscreteGraphicalModel{Graph{Undirected}, L}) where L
+  global_markov(graph(M))
+end
+
+###################################################################################
+#
+#       Directed Discrete Graphical Models
+#
+###################################################################################
+
+@doc raw"""
+    discrete_graphical_model(G::Graph{Directed}, states::Vector{Int}; q_var_name::String="q")
+
+The parametric statistical model associated to a directed acyclic graph.
+It contains a directed acylic graph `G`, a MarkovRing `S` where the vanishing ideal of the model naturally lives,
+and a parameter ring whose variables `q[i][j](j_1, j_2, ..., j_#pa(i))` correspond to the conditional distribution
+of the node i given its parents.
+
+``` jldoctest 
+julia> M = discrete_graphical_model(graph_from_edges(Directed, [[1,3], [2,3], [3,4]]), [2,2,2,2])
+Discrete Graphical Model on a Directed graph with 4 nodes and 3 edges with states [2, 2, 2, 2]
+
+```
+"""
+function discrete_graphical_model(G::Graph{Directed}, states::Vector{Int}; q_var_name::String="q", p_var_name::String="p")
+  @req length(states) == n_vertices(G) "need one and only one random variable in MarkovRing for each graph vertex"
+  @req is_acyclic(G) "$G must be an acyclic graph"
+  DiscreteGraphicalModel(G, states, Dict{Symbol, VarName}(:q => q_var_name, :p => p_var_name))
+end
+
+@doc raw"""
+  @attr Tuple{MPolyRing, GenDict} parameter_ring(M::DiscreteGraphicalModel{Graph{Directed}, T}; cached=false)
+
+Return the ring of parameters of the statistical model, together with a
+Dict for indexing its generators.
+
+There is a "hidden" homogenizing variable named "_h" in the ring. It can be
+obtained via `last(gens(R))` but is not listed in the dictionary of generators.
+
+``` jldoctest 
+julia> M = discrete_graphical_model(graph_from_edges(Directed, [[1,3], [2,3], [3,4]]), [2,2,2,2])
+Discrete Graphical Model on a Directed graph with 4 nodes and 3 edges with states [2, 2, 2, 2]
+
+julia> parameter_ring(M)[2]
+Dict{Tuple{Int64, Int64, Vector{Int64}}, QQMPolyRingElem} with 16 entries:
+  (4, 2, [1])    => q[4](2 | 1)
+  (3, 1, [2, 2]) => q[3](1 | 2, 2)
+  (2, 2, [])     => q[2](2)
+  (3, 1, [1, 2]) => q[3](1 | 2, 1)
+  (4, 1, [2])    => q[4](1 | 2)
+  (1, 1, [])     => q[1](1)
+  (3, 2, [2, 2]) => q[3](2 | 2, 2)
+  (3, 2, [1, 2]) => q[3](2 | 2, 1)
+  (4, 2, [2])    => q[4](2 | 2)
+  (1, 2, [])     => q[1](2)
+  (3, 1, [2, 1]) => q[3](1 | 1, 2)
+  (3, 1, [1, 1]) => q[3](1 | 1, 1)
+  (4, 1, [1])    => q[4](1 | 1)
+  (2, 1, [])     => q[2](1)
+  (3, 2, [2, 1]) => q[3](2 | 1, 2)
+  (3, 2, [1, 1]) => q[3](2 | 1, 1)
+```
+"""
+@attr Tuple{
+  MPolyRing,
+  GenDict
+} function parameter_ring(M::DiscreteGraphicalModel{Graph{Directed}, T}; cached=false) where T
+  G = graph(M)
+  params = []
```

Comment:
Type unstable. Should perhaps be something like this, where `TODO` is the type of what `state_space` elements look like?
```suggestion
  params = Tuple{Int, TODO, Vector{Int}}[]
```